src/main/java/org/opentripplanner/middleware/controllers/api/AdminUserController.java

```diff
@@ -0,0 +1,57 @@
+package org.opentripplanner.middleware.controllers.api;
+
+import com.auth0.exception.Auth0Exception;
+import com.auth0.json.mgmt.users.User;
+import org.opentripplanner.middleware.models.AdminUser;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+
+import static org.opentripplanner.middleware.auth.Auth0Users.deleteAuth0User;
+import static org.opentripplanner.middleware.auth.Auth0Users.updateAuthFieldsForUser;
+import static org.opentripplanner.middleware.auth.Auth0Users.createNewAuth0User;
+import static org.opentripplanner.middleware.auth.Auth0Users.validateExistingUser;
+import static org.opentripplanner.middleware.utils.JsonUtils.logMessageAndHalt;
+
+/**
+ * Implementation of the {@link ApiController} abstract class for managing users. This controller connects with Auth0
+ * services using the hooks provided by {@link ApiController}.
+ */
+public class AdminUserController extends ApiController<AdminUser> {
+    private static final Logger LOG = LoggerFactory.getLogger(AdminUserController.class);
+
+    public AdminUserController(String apiPrefix){
+        super(apiPrefix, Persistence.adminUsers, "admin/user");
```

Comment:
This super constructor here is really the only difference between the `AdminUserController`, `ApiUserController` and `UserController`. Perhaps the hooks should be moved into an `AbstractUserController` class or the `ApiController` class.

Reply:
Perhaps. Mind if we create a ticket and handle in another PR?

Reply:
ok